lib/ohai/plugins/c.rb

```diff
@@ -39,6 +40,7 @@
         end
       end
     rescue Errno::ENOENT
```

Comment:
Not always... :(

```
[5] pry(main)> so = Mixlib::ShellOut.new('cl /?').run_command
=> <Mixlib::ShellOut#70288827374520: command: 'cl /?' process_status: #<Process::Status: pid 65429 exit 127> stdout: '' stderr: 'sh: cl: command not found' child_pid: 65429 environment: {} timeout: 600 user:  group:  working_dir:  >
[6] pry(main)> so.exitstatus
=> 127
```

I'm really confused about this behavior in shell_out (raising errors vs. non-zero exit codes when a command isn't found).